packages/edit-post/src/components/manage-blocks-modal/manager.js

```diff
@@ -0,0 +1,193 @@
+/**
+ * External dependencies
+ */
+import { filter, includes, map, find, some, deburr } from 'lodash';
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { withSelect, withDispatch } from '@wordpress/data';
+import { compose, withInstanceId, withState } from '@wordpress/compose';
+import { ToggleControl, PanelBody } from '@wordpress/components';
+import { __experimentalBlockTypesList as BlockTypesList } from '@wordpress/editor';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Filters an item list given a search term.
+ *
+ * @param {Object[]} items      Set of inserter items to search within.
+ * @param {Object[]} categories Set of categories to search within.
+ * @param {string}   searchTerm Search term.
+ *
+ * @return {Object[]} Filtered item list.
+ */
+export const searchItems = ( items, categories, searchTerm ) => {
+	const normalizedSearchTerm = normalizeTerm( searchTerm );
+	const matchSearch = ( string ) => normalizeTerm( string ).includes( normalizedSearchTerm );
+
+	return items.filter( ( item ) => {
+		const itemCategory = find( categories, { slug: item.category } );
+
+		return (
+			matchSearch( item.title ) ||
+			some( item.keywords, matchSearch ) ||
+			( itemCategory && matchSearch( itemCategory.title ) )
+		);
+	} );
+};
+
+/**
+ * Converts the search term into a normalized term.
+ *
+ * @param {string} term The search term to normalize.
+ *
+ * @return {string} The normalized search term.
+ */
+export const normalizeTerm = ( term ) => {
+	// Disregard diacritics.
+	//  Input: "média"
+	term = deburr( term );
+
+	// Accommodate leading slash, matching autocomplete expectations.
+	//  Input: "/media"
+	term = term.replace( /^\//, '' );
+
+	// Lowercase.
+	//  Input: "MEDIA"
+	term = term.toLowerCase();
+
+	// Strip leading and trailing whitespace.
+	//  Input: " media "
+	term = term.trim();
+
+	return term;
+};
+
+function BlockManager( {
+	instanceId,
+	categories,
+	blockTypes,
+	disabledBlockTypes,
+	enableBlockTypes,
+	disableBlockTypes,
+	search,
+	setState,
+} ) {
+	const blockItems = blockTypes.map( ( blockType ) => ( {
+		id: blockType.name,
+		icon: blockType.icon,
+		title: blockType.title,
+		category: blockType.category,
+		hasChildBlocksWithInserterSupport: false,
+		className: classnames( 'edit-post-manage-blocks-modal__block-type', {
+			'is-inactive': includes( disabledBlockTypes, blockType.name ),
+		} ),
```

Comment:
So the difference with the blocks inserter is that the block that can't be inserted has `button` with `disabled` attribute and `aria-hidden` set to false for the icon. This hides it from interactions when using keyboard navigation (tabs). It behaves differently here because we still want to be able to navigate to the block item to be able to enable it. I think that this might be a challenge here because we want a block icon to behave like a checkbox. It should announce which state it is in and change it when toggled.

Reply:
This aligns what @mapk shared in his previous comment: https://github.com/WordPress/gutenberg/pull/14224#issuecomment-469864535.